extensions/miro/src/list-boards.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from "react";
+import * as miro from "./oauth/miro";
+import { Board } from "@mirohq/miro-api";
+import { Action, ActionPanel, Detail, Icon, List, showToast, Toast, useNavigation } from "@raycast/api";
+import InviteBoard from "./invite-board";
+import ListMembers from "./list-member";
+
+export default function ListBoards() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [items, setItems] = useState<Board[]>([]);
+
+  const { push } = useNavigation();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        await miro.authorize();
+        const fetchedItems = await miro.fetchItems();
+        setItems(fetchedItems);
+        setIsLoading(false);
+      } catch (error) {
+        console.error(error);
+        setIsLoading(false);
+        await showToast({ style: Toast.Style.Failure, title: String(error) });
+      }
+    })();
+  }, []);
+
+  if (isLoading) {
+    return <Detail isLoading={isLoading} />;
```

Comment:
I'd say that it would be nicer to return the list right away so that you can start typing in the search bar without waiting for the results

Reply:
Fixed this! 🙏